pkg/cmd/ui/tty.go

```diff
@@ -47,3 +47,14 @@ type noopWriter struct{}
 var _ io.Writer = noopWriter{}
 
 func (w noopWriter) Write(data []byte) (int, error) { return len(data), nil }
+
+// Used for testing whether TTY writes correct output to stdout/stderr
+func NewMockTTY(debug bool, stdout, stderr io.Writer) tty {
```

Comment:
in general public constructors should return public concrete types since it should be possible to describe returned type from the call site. e.g.

```go
var myFoo ____ = ui.NewMockTTY()
```

what is ____ above?

Reply:
Exported the TTY type. Was there a different concept name you prefer here? I am mainly following the change from #265.